yunwoo/programmers-2/할인-행사.js

```diff
@@ -0,0 +1,29 @@
+function solution(want, number, discount) {
+  let result = 0;
+
+  discount.forEach((_, idx) => {
+    const wantQuantityObj = {};
+    const tenDaysDiscountList = discount.slice(idx, idx + 10);
+    let isWantDiscount = true;
+
+    if (tenDaysDiscountList.length < 10) return;
+
+    tenDaysDiscountList.forEach((el) => (wantQuantityObj[el] = (wantQuantityObj[el] || 0) + 1));
+
+    for (let i = 0; i < want.length; i++) {
+      if (wantQuantityObj[want[i]] !== number[i]) return (isWantDiscount = false);
+    }
+
+    if (isWantDiscount) result++;
+  });
+
+  return result;
+}
```

Comment:
```suggestion
function solution(want, number, discount) {
  let result = 0;
    
  for (let i = 0; i < discount.length; i++) {
    const wantQuantityObj = {};
    const tenDaysDiscountList = discount.slice(i, i + 10);
    let isWantDiscount = true;
    
    if (tenDaysDiscountList.length < 10) break;
      
    tenDaysDiscountList.forEach((el) => (wantQuantityObj[el] = (wantQuantityObj[el] || 0) + 1));

    for (let i = 0; i < want.length; i++) {
      if (wantQuantityObj[want[i]] !== number[i]) {
          isWantDiscount = false;
          break;
      };
    }

    if (isWantDiscount) result++;      
  }

  return result;
}
```
- 풀이잘봤습니다!
- 아무래도 forEach문 내부에 return문이 있어서 어색함을 느꼈는데 아무래도 일반적인 for문의 break문을 의도하신거겠죠?
- 차라리 for문으로 break문을 사용하면 어떨까 싶습니다! 🙇‍♂️

Reply:
- 헉 맞네요 break으로 한번에 종료하면 됐었군요!! 감사합니다 👍